src/platform/riot/coap_udp_ip_channel.c

```diff
@@ -26,21 +26,31 @@ static void _CoapUdpIpChannel_update_state(CoapUdpIpChannel *self, NetworkChanne
   NetworkChannelState old_state = self->state;
 
   // Update the state of the channel to its new state
+  mutex_lock(&self->state_mutex);
   self->state = new_state;
+  mutex_unlock(&self->state_mutex);
 
   // Inform runtime about new state if it changed from or to NETWORK_CHANNEL_STATE_CONNECTED
   if ((old_state == NETWORK_CHANNEL_STATE_CONNECTED && new_state != NETWORK_CHANNEL_STATE_CONNECTED) ||
       (old_state != NETWORK_CHANNEL_STATE_CONNECTED && new_state == NETWORK_CHANNEL_STATE_CONNECTED)) {
     _env->platform->new_async_event(_env->platform);
   }
+
+  // Let connection thread evaluate new state of this channel
+  msg_t msg = {
```

Comment:
So this writes a "msg" to the network thread to notify that the state has been updated. This implies that both runtime and network thread might update the state, I think we discussed earlier to just have the network thread update the state, but this was perhaps only possible for the TcpIp channel?